src/main/java/com/example/ludogoriesoft/lukeriaerpapi/config/SecurityConfiguration.java

```diff
@@ -48,7 +48,11 @@ public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Excepti
                         "/api/v1/images/**",
                         "/api/v1/auth/**")
                 .permitAll()
-
+                .requestMatchers(HttpMethod.GET,"/api/v1/user/me").authenticated()
+                .requestMatchers(HttpMethod.GET,"/api/v1/user/{id}").authenticated()
+                .requestMatchers(HttpMethod.PUT,"/api/v1/user/authenticated/{id}").authenticated()
+                .requestMatchers(HttpMethod.GET,"/api/v1/user/ifPassMatch").authenticated()
+                .requestMatchers(HttpMethod.PUT,"/api/v1/user/change-pass").authenticated()
```

Comment:
You can change every thing with .requestMatchers("/api/v1/user/**").authenticated()

Reply:
but I think that only admin should be able to access this:@PutMapping("/{id}")
    public ResponseEntity<UserDTO> updateUser(@PathVariable("id") Long id, @Valid @RequestBody UserDTO userDTO, @RequestHeader("Authorization") String auth).  and this:@GetMapping
    public ResponseEntity<List<UserDTO>> getAllUsers()

Reply:
I think this methods should be only for admins:
                .requestMatchers(HttpMethod.GET,"/api/v1/user").hasAnyRole(ADMIN.name())
                .requestMatchers(HttpMethod.POST,"/api/v1/user").hasAnyRole(ADMIN.name())
                .requestMatchers(HttpMethod.PUT,"/api/v1/user/{id}").hasAnyRole(ADMIN.name())
                .requestMatchers(HttpMethod.DELETE,"/api/v1/user/{id}").hasAnyRole(ADMIN.name())

Reply:
ok you can do it this way -.requestMatchers("/api/v1/user/**").hasAnyRole(ADMIN.name()) (when you don't declare HTTPMETOD , default to all), also no need to declare url after "/user" use **